genie-core/src/test/groovy/com/netflix/genie/core/util/ExponentialBackOffTriggerSpec.groovy

```diff
@@ -0,0 +1,150 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.core.util
+
+import com.netflix.genie.test.categories.UnitTest
+import org.junit.experimental.categories.Category
+import org.springframework.scheduling.TriggerContext
+import spock.lang.Specification
+import spock.lang.Unroll
+
+@Category(UnitTest)
+class ExponentialBackOffTriggerSpec extends Specification {
+
+    @Unroll
+    def "NextExecutionTime with type #delayType"(
+            ExponentialBackOffTrigger.DelayType delayType,
+            long[] nextExpectedExecutions
+    ) {
+        long min = 1L
+        long max = 16L
+        float factor = 2.0f
+        TriggerContext triggerContext = Mock(TriggerContext) {
+            lastScheduledExecutionTime() >> new Date(100)
+            lastActualExecutionTime() >> new Date(200)
+            lastCompletionTime() >> new Date(300)
+        }
+
+        setup:
```

Comment:
For my own edification, what is the difference between the above block of code which sets up some variables and this block which explicitly is labeled as setup?

Reply:
Setup should include those lines..